### PR TITLE
Updated to zend-servicemanager v3

### PR DESCRIPTION
**Pull request by weierophinney**
- composer.json entry for zend-servicemanager using `dev-develop as 2.7.0`.
- Updated `AbstractConfigFactory` to follow new interface definition.
- Updated `ReaderPluginManager` and `WriterPluginManager` to follow changes in `AbstractPluginManager`; they aggregate invokables and merge them with incoming configuration before passing on to the parent constructor.
- Updated `Factory` to pass an empty service manager instance to plugin managers when lazy loading them.

### CODE REVIEW

**Comment by danizord** [comment]
:+1:

**Comment by weierophinney** [comment]
Coverage is less, as I marked a number of tests that feature the "Translator" processor as incomplete, until that component is updated for zend-servicemanager v3.

File: src/ReaderPluginManager.php [hunk 1]
```diff
@@ -26,25 +29,9 @@ class ReaderPluginManager extends AbstractPluginManager
...
-            // we're okay
-            return;
-        }
-
-        throw new Exception\InvalidArgumentException(sprintf(
-            'Plugin of type %s is invalid; must implement %s\Reader\ReaderInterface',
-            (is_object($plugin) ? get_class($plugin) : gettype($plugin)),
-            __NAMESPACE__
-        ));
+        $config = array_merge_recursive(['invokables' => $this->invokableClasses], $config);
```
Comment (weierophinney) [inline src/ReaderPluginManager.php]: This approach is necessary as the SM doesn't have an `$invokableClasses` property any longer; it instead munges invokables into factory and alias definitions. As such, the configuration needs to be passed in during instantiation.

File: src/ReaderPluginManager.php [hunk 2]
```diff
@@ -26,25 +29,9 @@ class ReaderPluginManager extends AbstractPluginManager
...
 
-    /**
-     * Validate the plugin
-     * Checks that the reader loaded is an instance of Reader\ReaderInterface.
-     *
-     * @param  Reader\ReaderInterface $plugin
-     * @return void
-     * @throws Exception\InvalidArgumentException if invalid
-     */
-    public function validatePlugin($plugin)
```
Comment (weierophinney) [inline src/ReaderPluginManager.php]: `validatePlugin()` is no longer necessary, as the `AbstractPluginManager` has an implementation that refers to the `$instanceOf` property for the most common use case of validating against a type.

File: src/Factory.php [hunk 3]
```diff
@@ -220,7 +220,7 @@ public static function setReaderPluginManager(ReaderPluginManager $readers)
     public static function getReaderPluginManager()
     {
         if (static::$readers === null) {
-            static::$readers = new ReaderPluginManager();
+            static::$readers = new ReaderPluginManager(new ServiceManager());
```
Comment (weierophinney) [inline src/Factory.php]: This is required, as abstract plugin manager implementations now require the parent service manager during construction. I did the same with the writer plugin manager as well.

**Comment by mwillbanks** [comment]
Missing the changelog file :)

**Comment by weierophinney** [comment]
changelog added, @mwillbanks . :)

**Pull request merged**